### PR TITLE
Add io.github.roseblume.rosary

### PR DESCRIPTION
**Pull request by RoseBlume**
### Please confirm your submission meets all the criteria

- [X ] Please describe your application briefly.
This application is a simple bible app designed with the goal of providing Christians with reliable sources to review so that they may come to their own conclusions about their faith.
- [ X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ X] My pull request follows the instructions at [App Submission][submission].
- [ X] I have [built][build] and tested the submission locally.
- [ X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/RoseBlume/Rosary**
- [ X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [ X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by hfiguiere** [comment]
This is missing:

> Please describe your application briefly.

File: org.bible.Rosary.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+id: org.bible.Rosary
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:1]: Is this really the domain of the developer? 

> The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines](https://docs.flathub.org/docs/for-app-authors/requirements#application-id) are followed.

The checklist isn't a CAPTCHA.

File: org.bible.Rosary.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+runtime: org.gnome.Platform
+runtime-version: '45'
+sdk: org.gnome.Sdk
+command: rosary
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+build-options:
+  no-debuginfo: true
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:11]:
```suggestion
```

File: org.bible.Rosary.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>org.bible.Rosary</id>
+    <launchable type="desktop-id">org.bible.Rosary.desktop</launchable>
+    <name>Rosary</name>
+    <developer_name>James A. Rose</developer_name>
+    <content_rating type="oars-1.0">
```
Comment (hfiguiere) [inline org.bible.Rosary.metainfo.xml:7]:
```suggestion
    <content_rating type="oars-1.1">
```

File: org.bible.Rosary.metainfo.xml
Comment (hfiguiere) [inline org.bible.Rosary.metainfo.xml:1]: this should be upstream.

File: org.bible.Rosary.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            A simple Bible app written with Tauri, using a Rust backend and a SolidJS frontend. This projects goal is to allow each person to study the bible and come to their own conclusion about their faith.
+        </p>
+    </description>
+    <screenshots>
+        <screenshot type="default">
+            <image>https://raw.githubusercontent.com/RoseBlume/Rosary/master/main.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://raw.githubusercontent.com/RoseBlume/Rosary/master/menu.png</image>
+        </screenshot>
```
Comment (hfiguiere) [inline org.bible.Rosary.metainfo.xml:36]: screenshots should NEVER link from a branch (`master` in that case)

File: org.bible.Rosary.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+id: org.bible.Rosary
+runtime: org.gnome.Platform
+runtime-version: '45'
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:3]: 46 is the current one

File: rosary.desktop
Comment (hfiguiere) [inline rosary.desktop:1]: this should be managed upstream as well

File: org.bible.Rosary.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    buildsystem: simple
+    build-commands:
+    - mkdir -p /app/rosary
+    - unzip rosary.zip
+    - install -Dm755 bin/rosary /app/bin/rosary
+    - install -Dm644 rosary.desktop /app/share/applications/org.bible.Rosary.desktop
+
+    - install -Dm644 share/icons/hicolor/128x128/apps/rosary.png /app/share/icons/hicolor/128x128/apps/org.bible.Rosary.png
+    - install -Dm644 share/icons/hicolor/256x256@2/apps/rosary.png /app/share/icons/hicolor/256x256@2/apps/org.bible.Rosary.png
+    - install -Dm644 share/icons/hicolor/32x32/apps/rosary.png /app/share/icons/hicolor/32x32/apps/org.bible.Rosary.png
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:23]: you should use `rename-icon` in the manifest instead.

File: org.bible.Rosary.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    - install -Dm644 org.bible.Rosary.metainfo.xml /app/share/metainfo/org.bible.Rosary.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/RoseBlume/Rosary/releases/download/flatpak/Rosary.zip
+        sha256: 279d7d6971d4b526ad34c0907ade69a060c775efd490878c1b692262413ef7a8
+        dest-filename: rosary.zip
+        only-arches: [x86_64]
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:30]: 1. why?
2. How will that work in the aarch64 package being built?

File: org.bible.Rosary.yaml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    - install -Dm755 bin/rosary /app/bin/rosary
+    - install -Dm644 rosary.desktop /app/share/applications/org.bible.Rosary.desktop
+
+    - install -Dm644 share/icons/hicolor/128x128/apps/rosary.png /app/share/icons/hicolor/128x128/apps/org.bible.Rosary.png
+    - install -Dm644 share/icons/hicolor/256x256@2/apps/rosary.png /app/share/icons/hicolor/256x256@2/apps/org.bible.Rosary.png
+    - install -Dm644 share/icons/hicolor/32x32/apps/rosary.png /app/share/icons/hicolor/32x32/apps/org.bible.Rosary.png
+    - install -Dm644 org.bible.Rosary.metainfo.xml /app/share/metainfo/org.bible.Rosary.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/RoseBlume/Rosary/releases/download/flatpak/Rosary.zip
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:27]: this is unversionned. it's a bad idea.

File: org.bible.Rosary.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+build-options:
+  no-debuginfo: true
+modules:
+  - name: rosary
+    buildsystem: simple
+    build-commands:
+    - mkdir -p /app/rosary
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:16]: this directory is never used.

File: org.bible.Rosary.yaml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    - unzip rosary.zip
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:17]: you should have used `type: archive` instead below.

File: rosary.desktop [hunk 11]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Categories=Network;WebBrowser;
+Comment=A Bible app written with Tauri
```
Comment (hfiguiere) [inline rosary.desktop:3]: this is very poor "comment". I'm sure 100% of the user won't care about the tech stack. It's irrelevant. Same apply to the appstream file.

File: org.bible.Rosary.metainfo.xml
Comment (hfiguiere) [reply org.bible.Rosary.metainfo.xml:1]: also missing mandatory `releases`

File: org.bible.Rosary.metainfo.xml [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    <launchable type="desktop-id">org.bible.Rosary.desktop</launchable>
+    <name>Rosary</name>
+    <developer_name>James A. Rose</developer_name>
+    <content_rating type="oars-1.0">
+        <content_attribute id="violence-bloodshed">mild</content_attribute>
+    </content_rating>
+    <recommends>
+        <display_length compare="ge">360</display_length>
+    </recommends>
+    <summary>A simple Bible app written with Tauri, using a Rust backend and a SolidJS frontend.</summary>
```
Comment (hfiguiere) [inline org.bible.Rosary.metainfo.xml:13]:
```suggestion
    <summary>A simple Bible app</summary>
```

100% of the user don't care about the tech stack.

File: org.bible.Rosary.metainfo.xml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+
+    <supports>
+        <control>pointing</control>
+        <control>keyboard</control>
+        <control>touch</control>
+    </supports>
+
+    <description>
+        <p>
+            A simple Bible app written with Tauri, using a Rust backend and a SolidJS frontend. This projects goal is to allow each person to study the bible and come to their own conclusion about their faith.
```
Comment (hfiguiere) [inline org.bible.Rosary.metainfo.xml:27]:
```suggestion
            A simple Bible app. This projects goal is to allow each person to study the bible and come to their own conclusion about their faith.
```

File: org.bible.Rosary.yaml (hunk 1 above)
Comment (RoseBlume) [reply org.bible.Rosary.yaml]: I have fixed this it is now proper.

File: org.bible.Rosary.yaml [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+modules:
+  - name: rosary
+    buildsystem: simple
+    build-commands:
+    - ls
+    - ls icons/hicolor
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:15]:
```suggestion
```

File: org.bible.Rosary.yaml [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+  - name: rosary
+    buildsystem: simple
+    build-commands:
+    - install -Dm755 rosary /app/bin/rosary
+    - install -Dm644 applications/io.github.roseblume.rosary.desktop /app/share/applications/io.github.roseblume.rosary.desktop
+    - install -Dm644 icons/hicolor/scalable/apps/io.github.roseblume.rosary.svg /app/share/icons/hicolor/scalable/apps/io.github.roseblume.rosary.svg
+    - install -Dm644 metainfo/io.github.roseblume.rosary.xml /app/share/metainfo/io.github.roseblume.rosary.metainfo.xml
+    sources:
+      - type: archive
+        url: https://github.com/RoseBlume/Rosary/releases/download/v1.2.3/Rosary_1.2.3.zip
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:20]: how is this gonna work on multiple architectures? I see no compilation.

File: org.bible.Rosary.yaml [hunk 16]
```diff
@@ -0,0 +1,22 @@
+id: io.github.roseblume.rosary
+runtime: org.gnome.Platform
+runtime-version: '46'
+sdk: org.gnome.Sdk
+command: rosary
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
```
Comment (hfiguiere) [inline org.bible.Rosary.yaml:9]: --share=ipc is missing

File: org.bible.Rosary.yaml (hunk 15 above)
Comment (RoseBlume) [reply org.bible.Rosary.yaml:20]: This is intended to only be available on x86_64 and aarch64 architectures. I cannot find anything about specifiying architectures in the references. If you could guide me in the right direction that would be appreciated.

Comment (hfiguiere) [reply org.bible.Rosary.yaml:20]:
```
$ file bin/rosary 
bin/rosary: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=5e891b159f4eeaeedf9c793d7eb3f8cb703ea39e, for GNU/Linux 3.2.0, stripped
```

how is this gonna work on aarch64

Comment (hfiguiere) [reply org.bible.Rosary.yaml:20]: why not building from source? There is no cross compilation necessary.

Comment (RoseBlume) [reply org.bible.Rosary.yaml:20]: The compiling time, and the ram usage during compilation.

Comment (hfiguiere) [reply org.bible.Rosary.yaml:20]: and? how is that a problem?

Comment (RoseBlume) [reply org.bible.Rosary.yaml:20]: It wouldn't be able to be compiled.

Comment (hfiguiere) [reply org.bible.Rosary.yaml:20]: explain why it wouldn't?

Comment (RoseBlume) [reply org.bible.Rosary.yaml]: The make dependencies wouldn't be able to be satisfied. It would be easier to precompile it.

Comment (hfiguiere) [reply org.bible.Rosary.yaml]: again explain why.

Comment (RoseBlume) [reply org.bible.Rosary.yaml]: It runs a rust backend and a node front end. I would need to have npm, rustup, and I would not be able to remove the gnome sdk.

Comment (RoseBlume) [reply org.bible.Rosary.yaml]: I have setup aarch64 so it should work now.

Comment (hfiguiere) [reply org.bible.Rosary.yaml]: that still doesn't explain why it wouldn't build on flathub.

Comment (RoseBlume) [reply org.bible.Rosary.yaml]: I cannot install the proper tools on flathub.

Comment (RoseBlume) [reply org.bible.Rosary.yaml]: I got it working on aarch64 so it should be fine anyways

**Review by RoseBlume — COMMENTED** [review]
The Document has been corrected

**Comment by hfiguiere** [comment]
bot, build io.github.roseblume.rosary

File: org.bible.Rosary.metainfo.xml
Comment (hfiguiere) [reply org.bible.Rosary.metainfo.xml:1]: > also missing mandatory `releases`

this was ignored. that's the build failure

**Comment by RoseBlume** [comment]
Old changes have been re added. This has been tested an should work.

**Review by RoseBlume — COMMENTED** [review]
This includes fixes that were made previously but were undone in the previous commit.

**Pull request closed without merging**